sdks/java/extensions/sql/zetasql/src/main/java/org/apache/beam/sdk/extensions/sql/zetasql/ZetaSQLQueryPlanner.java

```diff
@@ -174,29 +174,20 @@ static boolean hasOnlyJavaUdfInProjects(RelOptRuleCall x) {
           } else if (rexNode instanceof RexLiteral) {
             SqlTypeName typeName = ((RexLiteral) rexNode).getTypeName();
             switch (typeName) {
-              case NULL:
-              case BOOLEAN:
-              case CHAR:
-              case BINARY:
-              case DECIMAL:
-                break;
-              default:
-                // Reject unsupported literals
+              case BIGINT: // BEAM-11989
+              case DATE: // BEAM-11990
                 return false;
+              default:
+                break;
             }
           } else if (rexNode instanceof RexInputRef) {
             SqlTypeName typeName = ((RexInputRef) rexNode).getType().getSqlTypeName();
             switch (typeName) {
-              case BIGINT:
-              case DOUBLE:
-              case BOOLEAN:
-              case VARCHAR:
-              case VARBINARY:
-              case DECIMAL:
-                break;
-              default:
-                // Reject unsupported input ref
+              case BIGINT: // BEAM-11989
```

Comment:
BigInt was actually in the supported types for InputRef before...

Reply:
I noticed that, but I wasn't sure why it is safe to include in input refs but not literals?

Reply:
Literals and input refs have a completely different implementation. BigInt is represented as a `BigInteger` in InputRef and a `BigDecimal` in Literals. (That is possibly a bug in Calcite.)